daemon/container_unix.go

```diff
@@ -1118,12 +1130,17 @@ func (container *Container) unmountVolumes(forceSyscall bool) error {
 func (container *Container) networkMounts() []execdriver.Mount {
 	var mounts []execdriver.Mount
 	shared := container.hostConfig.NetworkMode.IsContainer()
+	rootUID, rootGID := container.daemon.GetRemappedUIDGID()
 	if container.ResolvConfPath != "" {
 		label.Relabel(container.ResolvConfPath, container.MountLabel, shared)
 		writable := !container.hostConfig.ReadonlyRootfs
 		if m, exists := container.MountPoints["/etc/resolv.conf"]; exists {
 			writable = m.RW
 		}
+		// if we are going to mount a resolv.conf from container metadata
+		// we need to verify it is owned properly to support a daemon
+		// with user namespaces enabled
+		os.Chown(container.ResolvConfPath, rootUID, rootGID)
```

Comment:
ah--due to code moving around I had to change this in a recent rebase, and in this function there is no error path.  We can either warn/log error, or probably reasonable to change function defn. to return an error.  The only side effect of `chown` failure is that network files won't be modifiable by the root inside the user namespace'd container if this doesn't work, but that's annoying enough to error out.  I'll put in on my list to fix.